sdks/java/core/src/main/java/org/apache/beam/sdk/util/FileIOChannelFactory.java

```diff
@@ -73,8 +74,7 @@ private static File specToFile(String spec) {
 
     File parent = file.getAbsoluteFile().getParentFile();
     if (!parent.exists()) {
-      throw new FileNotFoundException(
-          "Parent directory " + parent + " of " + spec + " does not exist");
+      return Collections.EMPTY_LIST;
```

Comment:
I'm surprised this doesn't come with test changes. Add a test? Also, update the javadoc comment.